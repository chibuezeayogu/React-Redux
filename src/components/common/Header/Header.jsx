import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';

/**
 * @class Header
 * @extends Component
 */
export class Header extends Component {
	render() {
		const { courses, authors } = this.props;
 
		return (
			<nav className="navbar navbar-expand-lg navbar-light bg-light">
				<div className="container">
					<a className="navbar-brand">Navbar</a>
					<button
						className="navbar-toggler" 
						type="button" 
						data-toggle="collapse" 
						data-target="#navbarNav" 
						aria-controls="navbarNav" 
						aria-expanded="false" 
						aria-label="Toggle navigation"
					>
						<span className="navbar-toggler-icon"></span>
					</button>
					<div 
						className="collapse navbar-collapse justify-content-end"
						id="navbarNav">
						<div className="navbar-nav">
							<NavLink to="/"
								activeClassName="active"
								className="nav-item nav-link"
							>
									Home<span className="sr-only">(current)</span>
							</NavLink>
							<NavLink to="/courses"
								activeClassName="active"
								className="nav-item nav-link"
							>
									Courses <span className="badge badge-info">{courses.length}</span>
							</NavLink>
							<NavLink
								to="/about"
								activeClassName="active"
								className="nav-item nav-link"
							>
									About
							</NavLink>
						</div>
					</div>
				</div>
			</nav>
		);
	}
}

const mapStateTpProps = ({ coursesReducer, authorsReducer }) => ({
	courses: coursesReducer.courses,
	authors: authorsReducer.authors
});

export default connect(mapStateTpProps, null)(Header);
