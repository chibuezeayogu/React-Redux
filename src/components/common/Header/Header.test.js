import React from 'react';
import { shallow } from 'enzyme';

import { Header } from './Header';


describe('Header Component', () => {
	const props = {
		courses: [{
			id: 'react-flux-building-applications',
			title: 'Building Applications in React and Flux',
			watchHref: 'http://www.pluralsight.com/courses/react-flux',
			authorId: 'cory-house',
			length: '5:08',
			category: 'JavaScript'
		}]
	};

	const wrapper = shallow(<Header {...props} />);
  
	it('should render correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
