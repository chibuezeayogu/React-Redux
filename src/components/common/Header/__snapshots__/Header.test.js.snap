// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
    >
      Navbar
    </a>
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarNav"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse justify-content-end"
      id="navbarNav"
    >
      <div
        className="navbar-nav"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-item nav-link"
          to="/"
        >
          Home
          <span
            className="sr-only"
          >
            (current)
          </span>
        </NavLink>
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-item nav-link"
          to="/courses"
        >
          Courses
        </NavLink>
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav-item nav-link"
          to="/about"
        >
          About
        </NavLink>
      </div>
    </div>
  </div>
</nav>
`;
